api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -409,14 +409,10 @@ public void saveEncounter_shouldCascadeSaveToContainedConditions() {
 		Context.getEncounterService().saveEncounter(encounter);
 		
 		// verify
-		Object[] conditionsArray = encounter.getConditions().toArray();
-		pregnancy = (Condition) conditionsArray[0];
-		edema = (Condition) conditionsArray[1];
-		assertEquals(2, conditionsArray.length);
-		assertNotNull(pregnancy.getId());
-		assertEquals("Pregnancy", pregnancy.getCondition().getNonCoded());
-		assertNotNull(edema.getId());
-		assertEquals("Edema", edema.getCondition().getNonCoded());
+		Set<Condition> savedConditions = encounter.getConditions();
```

Comment:
I'd honestly feel more comfortable with the changes here if we can prove that the same test still passes or, if it doesn't, that the reason it doesn't pass is because of an expected change in behaviour.

Reply:
@ibacher I had the same concern yesterday. But looking closer, it appears that there is no specific care being given at the order in which those conditions are returned (before or after this PR). So I suggested to @icrc-loliveira to simply test that those two conditions were part of the returned set.

Reply:
@mks-d That's fair. My main worry was that the condition non-coded value might not be being preserved (since that's no longer verified). Order-wise, it's probably not too important.